src/Workspaces/Core/Portable/Workspace/Solution/DocumentState.cs

```diff
@@ -19,6 +19,8 @@ namespace Microsoft.CodeAnalysis
 {
     internal partial class DocumentState : TextDocumentState
     {
+        private static readonly Func<string, PreservationMode, string> s_fullParseLog = (path, mode) => $"{path} : {mode}";
```

Comment:
Since 'mode' has only two values you could do:

```(path, mode) => path + " : " + (mode == PreservationMode.PreserveValue) ? "PreserveValue" : "PreserveIdentity");```